### PR TITLE
Add support for Swarm node content in ContainerInfo response

### PR DESCRIPTION
**Pull request by itzg**
This is a rebase of #310 since #313 covers the label filtering part.

### CODE REVIEW

File: src/main/java/com/spotify/docker/client/messages/ContainerInfo.java [hunk 1]
```diff
@@ -53,6 +53,7 @@
   @JsonProperty("MountLabel") private String mountLabel;
   @JsonProperty("Volumes") private ImmutableMap<String, String> volumes;
   @JsonProperty("VolumesRW") private ImmutableMap<String, Boolean> volumesRW;
+  @JsonProperty("Node") private Node node;
```
Comment (davidxia) [inline src/main/java/com/spotify/docker/client/messages/ContainerInfo.java]: Maybe add a comment here that this field is only used when using docker swarm which extends existing docker API behavior?

Comment (itzg) [inline src/main/java/com/spotify/docker/client/messages/ContainerInfo.java]: I was considering that, but there wasn't a precedent for comments elsewhere in this file ;) Will do.

File: src/main/java/com/spotify/docker/client/messages/ContainerInfo.java [hunk 2]
```diff
@@ -252,6 +265,64 @@ public String toString() {
...
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) {
+        return true;
+      }
+      if (o == null || getClass() != o.getClass()) {
+        return false;
+      }
+      Node node = (Node) o;
+      return com.google.common.base.Objects.equal(id, node.id) &&
```
Comment (davidxia) [inline src/main/java/com/spotify/docker/client/messages/ContainerInfo.java]: Can we use `import com.google.common.base.Objects;` for these?

**Comment by davidxia** [comment]
LGTM other than my comment and the checkstyle error.

File: src/main/java/com/spotify/docker/client/messages/ContainerInfo.java (hunk 2 above)
Comment (itzg) [inline src/main/java/com/spotify/docker/client/messages/ContainerInfo.java]: I'll blame IntelliJ for that one. New commit in just a sec.

**Pull request merged**